src/main/kotlin/org/jetbrains/research/testspark/tools/llm/generation/hf/HuggingFaceRequestManager.kt

```diff
@@ -0,0 +1,116 @@
+import com.google.gson.JsonParser
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.HttpRequests
+import com.intellij.util.io.HttpRequests.HttpStatusException
+import org.jetbrains.research.testspark.bundles.llm.LLMDefaultsBundle
+import org.jetbrains.research.testspark.bundles.llm.LLMMessagesBundle
+import org.jetbrains.research.testspark.core.data.ChatUserMessage
+import org.jetbrains.research.testspark.core.monitor.ErrorMonitor
+import org.jetbrains.research.testspark.core.progress.CustomProgressIndicator
+import org.jetbrains.research.testspark.core.test.TestsAssembler
+import org.jetbrains.research.testspark.tools.llm.SettingsArguments
+import org.jetbrains.research.testspark.tools.llm.error.LLMErrorManager
+import org.jetbrains.research.testspark.tools.llm.generation.IJRequestManager
+import org.jetbrains.research.testspark.tools.llm.generation.JUnitTestsAssembler
+import java.net.HttpURLConnection
+
+/**
+ * A class to manage requests sent to large language models hosted on HuggingFace
+ */
+class HuggingFaceRequestManager(project: Project) : IJRequestManager(project) {
+    private val url = "https://api-inference.huggingface.co/models/meta-llama/"
+
+    // TODO: The user should be able to change these numbers in the plugin's settings
+    private val topProbability = 0.9
+    private val temperature = 0.9
+
+    private val llmErrorManager = LLMErrorManager()
+
+    override fun send(
+        prompt: String,
+        indicator: CustomProgressIndicator,
+        testsAssembler: TestsAssembler,
+        errorMonitor: ErrorMonitor,
+    ): SendResult {
+        val httpRequest = HttpRequests.post(
+            url + SettingsArguments(project).getModel(),
+            "application/json",
+        ).tuner {
+            it.setRequestProperty("Authorization", "Bearer $token")
+        }
+
+        // Add system prompt
+        if (chatHistory.size == 1) {
+            chatHistory[0] = ChatUserMessage(
+                createInstructionPrompt(
+                    chatHistory[0].content,
+                ),
+            )
+        }
+
+        val llmRequestBody = HuggingFaceRequestBody(chatHistory, Parameters(topProbability, temperature)).toMap()
+        var sendResult = SendResult.OK
+        try {
+            httpRequest.connect {
+                it.write(GsonBuilder().disableHtmlEscaping().create().toJson(llmRequestBody))
+                when (val responseCode = (it.connection as HttpURLConnection).responseCode) {
+                    HttpURLConnection.HTTP_OK -> {
+                        val text = it.reader.readLine()
+                        val generatedTestCases = extractLLMGeneratedCode(
+                            JsonParser.parseString(text).asJsonArray[0]
+                                .asJsonObject["generated_text"].asString.trim(),
+                        )
+                        (testsAssembler as JUnitTestsAssembler).consume(generatedTestCases)
+                    }
```

Comment:
Since we use the method coming from the interface, it can be simply:
```java
testsAssembler.consume(generatedTestCases)
```
No cast is needed; it is even required to remove the cast since other languages, such as Kotlin, will be coming soon.

Reply:
Addressed in https://github.com/JetBrains-Research/TestSpark/pull/212/commits/e218dfa573d4bd55d4ede21efdc17edcac461852.